### ISSUE TITLE
Build failure with HDF5 1.8.15pre7

### ISSUE THREAD

**Issue by avalentino** [PyTables/PyTables#443]
```
$ env HDF5_DIR=$HOME/hdf5-1.8.15/ python3 setup.py build
* Using Python 3.4.3 (default, Mar 26 2015, 22:03:40) 
* Found HDF5 headers at ``/home/antonio/hdf5-1.8.15/include``, library at ``/home/antonio/hdf5-1.8.15/lib``.
.. WARNING:: Could not find the HDF5 runtime.
   The HDF5 shared library was *not* found in the default library
   paths. In case of runtime problems, please remember to install it.
* Found LZO 2 headers at ``/usr/include``, the library is located in the standard system search dirs.
* Skipping detection of LZO 1 since LZO 2 has already been found.
* Found bzip2 headers at ``/usr/include``, the library is located in the standard system search dirs.
* Could not find blosc headers and library; using internal sources.
Compiling tables/utilsextension.pyx because it changed.
Cythonizing tables/utilsextension.pyx
warning: tables/utilsextension.pyx:804:12: Non-trivial type declarators in shared declaration (e.g. mix of pointers and values). Each pointer declaration should be on its own line.
warning: tables/utilsextension.pyx:804:22: Non-trivial type declarators in shared declaration (e.g. mix of pointers and values). Each pointer declaration should be on its own line.
Compiling tables/hdf5extension.pyx because it changed.
Cythonizing tables/hdf5extension.pyx
Compiling tables/tableextension.pyx because it changed.
Cythonizing tables/tableextension.pyx
Compiling tables/linkextension.pyx because it changed.
Cythonizing tables/linkextension.pyx
Compiling tables/_comp_lzo.pyx because it changed.
Cythonizing tables/_comp_lzo.pyx
Compiling tables/_comp_bzip2.pyx because it changed.
Cythonizing tables/_comp_bzip2.pyx
Compiling tables/lrucacheextension.pyx because it changed.
Cythonizing tables/lrucacheextension.pyx
Compiling tables/indexesextension.pyx because it changed.
Cythonizing tables/indexesextension.pyx
Setting compiler flag: -march=native
running build
running build_py
creating build
creating build/lib.linux-x86_64-3.4

[...]

x86_64-linux-gnu-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -g -fstack-protector-strong -Wformat -Werror=format-security -D_FORTIFY_SOURCE=2 -fPIC -DNDEBUG=1 -DHAVE_LZO2_LIB=1 -DHAVE_BZ2_LIB=1 -DHAVE_LZ4=1 -DHAVE_SNAPPY=1 -DHAVE_ZLIB=1 -Ic-blosc/hdf5 -I/home/antonio/hdf5-1.8.15/include -Ic-blosc/blosc -Ic-blosc/internal-complibs/zlib-1.2.8 -Ic-blosc/internal-complibs/snappy-1.1.1 -Ic-blosc/internal-complibs/lz4-r119 -I/usr/lib/python3/dist-packages/numpy/core/include -I/usr/include/python3.4m -c src/H5ATTR.c -o build/temp.linux-x86_64-3.4/src/H5ATTR.o -Isrc -DH5Acreate_vers=2 -DH5Aiterate_vers=2 -DH5Dcreate_vers=2 -DH5Dopen_vers=2 -DH5Eclear_vers=2 -DH5Eprint_vers=2 -DH5Epush_vers=2 -DH5Eset_auto_vers=2 -DH5Eget_auto_vers=2 -DH5Ewalk_vers=2 -DH5E_auto_t_vers=2 -DH5Gcreate_vers=2 -DH5Gopen_vers=2 -DH5Pget_filter_vers=2 -DH5Pget_filter_by_id_vers=2 -DH5Tarray_create_vers=2 -DH5Tget_array_dims_vers=2 -DH5Z_class_t_vers=2 -march=native
x86_64-linux-gnu-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -g -fstack-protector-strong -Wformat -Werror=format-security -D_FORTIFY_SOURCE=2 -fPIC -DNDEBUG=1 -DHAVE_LZO2_LIB=1 -DHAVE_BZ2_LIB=1 -DHAVE_LZ4=1 -DHAVE_SNAPPY=1 -DHAVE_ZLIB=1 -Ic-blosc/hdf5 -I/home/antonio/hdf5-1.8.15/include -Ic-blosc/blosc -Ic-blosc/internal-complibs/zlib-1.2.8 -Ic-blosc/internal-complibs/snappy-1.1.1 -Ic-blosc/internal-complibs/lz4-r119 -I/usr/lib/python3/dist-packages/numpy/core/include -I/usr/include/python3.4m -c c-blosc/hdf5/blosc_filter.c -o build/temp.linux-x86_64-3.4/c-blosc/hdf5/blosc_filter.o -Isrc -DH5Acreate_vers=2 -DH5Aiterate_vers=2 -DH5Dcreate_vers=2 -DH5Dopen_vers=2 -DH5Eclear_vers=2 -DH5Eprint_vers=2 -DH5Epush_vers=2 -DH5Eset_auto_vers=2 -DH5Eget_auto_vers=2 -DH5Ewalk_vers=2 -DH5E_auto_t_vers=2 -DH5Gcreate_vers=2 -DH5Gopen_vers=2 -DH5Pget_filter_vers=2 -DH5Pget_filter_by_id_vers=2 -DH5Tarray_create_vers=2 -DH5Tget_array_dims_vers=2 -DH5Z_class_t_vers=2 -march=native
c-blosc/hdf5/blosc_filter.c: In function ‘blosc_filter’:
c-blosc/hdf5/blosc_filter.c:201:13: error: format not a string literal and no format arguments [-Werror=format-security]
             PUSH_ERR("blosc_filter", H5E_CALLBACK, errmsg);
             ^
cc1: some warnings being treated as errors
error: command 'x86_64-linux-gnu-gcc' failed with exit status 1
```

The issue seems to be related to blosc but I have still not had the time to investigate.

Not sure if this is blocking for 3.2, probably yes.

**Comment by FrancescAlted** [comment]
Uh, I have never seen this error before.  Maybe this is due to a 'feature' of HDF5 1.8.15?

**Comment by avalentino** [comment]
> Uh, I have never seen this error before. Maybe this is due to a 'feature' of HDF5 1.8.15?

Probably, or maybe a bug in the pre release.
Still didn't looked at it.

**Comment by andreabedini** [comment]
It appears to be because of this http://stackoverflow.com/a/4419347, i.e. you not supposed to pass to `printf` something that is not a string literal.

**Comment by avalentino** [comment]
Yes, Debian enforces this kind of checks and converts warnings in errors.

**Comment by FrancescAlted** [comment]
Should have been fixed in fe92bb9
